### PR TITLE
Use redirectCount set through managerModifyRequest

### LINKED ISSUE #243: Setting redirectCount to 0 with managerModifyRequest does not work
I am using stackage's lts-7.3 which pulls http-client 0.4.31.1 by default and setting redirectCount to 0 does not work as expected: Redirection is still followed. I can see this behaviour has been fixed by https://github.com/snoyberg/http-client/commit/11a9a6273375e67c157b4b0a8a359c73029512cb Would it be possible to backport it so that it is available using lts-7.X dependency set?

### PR DESCRIPTION
**Pull request by abailly**
fixes #243 for maintenance branch.

### CODE REVIEW

File: http-client/Network/HTTP/Client.hs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {-# LANGUAGE DeriveFunctor #-}
-{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DeriveTraversable  #-}
```
Comment (snoyberg) [inline http-client/Network/HTTP/Client.hs]: Unintended change?

File: http-client/test/Network/HTTP/ClientSpec.hs [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {-# LANGUAGE OverloadedStrings #-}
 module Network.HTTP.ClientSpec where
 
-import           Control.Exception         (toException)
-import           Network                   (withSocketsDo)
-import           Network.HTTP.Client
-import           Network.HTTP.Types        (status200, status405)
-import           Test.Hspec
-import           Data.ByteString.Lazy.Char8 () -- orphan instance
```
Comment (snoyberg) [inline http-client/test/Network/HTTP/ClientSpec.hs]: Were these changes needed?

File: http-client/Network/HTTP/Client.hs (hunk 1 above)
Comment (abailly) [inline http-client/Network/HTTP/Client.hs]: Yeah, sorry for that. I had my stylish-haskell on, it was late, I tried to fix it manually... I will do another PR to remove that.

**Comment by snoyberg** [comment]
Not a big deal tbh.

**Comment by abailly** [comment]
I know, but I also know it adds unwanted noise down the line :)

**Pull request merged**